experimental/devices/inky/inky.go

```diff
@@ -0,0 +1,291 @@
+	Black:  0x00,
+	Red:    0x33,
+	Yellow: 0x33,
+	White:  0xff,
+}
+
+// NewpHAT opens a handle to an Inky pHAT.
+func NewpHAT(p spi.Port, dc gpio.PinOut, reset gpio.PinOut, busy gpio.PinIn, color Color) (*Dev, error) {
+	if color != Black && color != Red && color != Yellow {
+		return nil, fmt.Errorf("Unsupported color: %v", color)
+	}
+
+	c, err := p.Connect(speed, spi.Mode0, spiBits)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to inky over spi: %v", err)
+	}
+
+	d := &Dev{
+		c:      c,
+		dc:     dc,
+		r:      reset,
+		busy:   busy,
+		color:  color,
+		border: Black,
+	}
+
+	return d, nil
+}
+
+// Dev is a handle to an Inky.
+type Dev struct {
+	c conn.Conn
+	// Low when sending a command, high when sending data.
+	dc gpio.PinOut
+	// Reset pin, active low.
+	r gpio.PinOut
+	// High when device is busy.
+	busy gpio.PinIn
+
+	// Color of device screen (red, yellow or black).
+	color Color
+	// Modifiable color of border.
+	border Color
+}
+
+// SetBorder changes the border color. This will not take effect until the next Draw().
+func (d *Dev) SetBorder(c Color) {
+	d.border = c
+}
+
+// String implements conn.Resource.
+func (d *Dev) String() string {
+	return "InkyPHat"
+}
+
+// Halt implements conn.Resource
+func (d *Dev) Halt() error {
+	return nil
+}
+
+// ColorModel implements display.Drawer
+// Maps white to white, black to black and anything else as red. Red is used as
+// a placeholder for the display's third color, i.e., red or yellow.
+func (d *Dev) ColorModel() color.Model {
+	return color.ModelFunc(func(c color.Color) color.Color {
+		r, g, b, _ := c.RGBA()
+		if r == 0 && g == 0 && b == 0 {
+			return color.RGBA{
+				R: 0,
+				G: 0,
+				B: 0,
+				A: 255,
+			}
+		} else if r == 0xffff && g == 0xffff && b == 0xffff {
+			return color.RGBA{
+				R: 255,
+				G: 255,
+				B: 255,
+				A: 255,
+			}
+		}
+		return color.RGBA{
+			R: 255,
+			G: 0,
+			B: 0,
+			A: 255,
+		}
+	})
+}
+
+// Bounds implements display.Drawer
+func (d *Dev) Bounds() image.Rectangle {
+	return image.Rect(0, 0, rows, cols)
+}
+
+// Draw implements display.Drawer
+func (d *Dev) Draw(dstRect image.Rectangle, src image.Image, srcPtrs image.Point) error {
+	if dstRect != d.Bounds() {
+		return fmt.Errorf("Partial update not supported")
+	}
+
+	if src.Bounds() != d.Bounds() {
+		return fmt.Errorf("Image must be the same size as bounds: %v", d.Bounds())
+	}
+
+	b := src.Bounds()
+	// Black/white pixels.
+	white := make([]bool, rows*cols)
+	// Red/Transparent pixels.
+	red := make([]bool, rows*cols)
+	for x := b.Min.X; x < b.Max.X; x++ {
+		for y := b.Min.Y; y < b.Max.Y; y++ {
+			i := x*cols + y
+			srcX := x
+			srcY := b.Max.Y - y - 1
+			r, g, b, _ := d.ColorModel().Convert(src.At(srcX, srcY)).RGBA()
+			if r == 0xffff && g == 0xffff && b == 0xffff {
+				white[i] = true
+			} else if r == 0xffff {
+				// Red pixels also need white behind them.
+				white[i] = true
+				red[i] = true
+			}
+		}
+	}
+
+	bufA, _ := pack(white)
```

Comment:
Not required, but personally I'd recommend to let go the []bool and immediately work in packed data instead. It would look a bit like this:

```
index := x / 8 + y * b.Max.X
shift := 7 - uint(x)%8
ret[index] |= 1 << shift
```

but you only need to do these for the `true` assignments, removing the need for boolToByte.